src/name/verify.rs

```diff
@@ -294,9 +294,16 @@ enum NameIteration {
     Stop(Result<(), Error>),
 }
 
+#[derive(Clone, Copy)]
+pub(crate) enum TreatSubjectCommonNameAsDnsName {
```

Comment:
This feels like an unidiomatically closed formulation of what we need. Could we respin it as something more like `CommonNameType { Dns, Any }` or `CommonNameType { Dns, Other }`? Or maybe even `SubjectType { ServerEndEntity, Other }`?